src/analytics.ts

```diff
@@ -41,14 +42,15 @@ export default class Analytics {
     );
   }
 
-  private userID(): string | undefined {
+  private userID(): string {
     const KEY = 'userLogID';
-    if (this.store.get(KEY) == undefined) {
+    const value = this.store.get<string>(KEY);
+    if (isUndefined(value)) {
       const newID = uuid.uuid();
       void this.store.update(KEY, newID);
       return newID;
     } else {
-      return this.store.get(KEY);
+      return value;
```

Comment:
We have established that `value` is not undefined at this point and the other conditional branch also returns a string and so we have removed the possibility of undefined.